### PR TITLE
VACMS-1798: Fix Outreach Hub Events listing access.

### PR DESCRIPTION
**Pull request by oksana-c**
## Description

See #1798. 

## Testing done
Manual

## Screenshots

Screen Shot 2020-05-20 at 12 10 43 PM

**screenshots for engineers to validate access assumptions**
Screen Shot 2020-05-20 at 11 51 02 AM
Screen Shot 2020-05-20 at 11 52 54 AM
^ odd that the column storing term id is actually named "section_id"...

## QA steps

**Assumption: users who have access to the following sections should be able to create events in Outreach Hub:**
* `- Veterans Affairs (tid:5 | section_id:7)`
* `-- Office of Public & Intergovernmental Affairs (tid:6 | section_id:9)`
* `--- Outreach Hub (tid:7 | section_id:4)`

Login as admin and assign "Content Editor - VAMC" role to `tucker.patterson@va.gov`

Login as `tucker.patterson@va.gov`. Verify that Tucker is assigned to the Outreach Hub section.
- [x] create an Event - http://pr1800.ci.cms.va.gov/node/add/event and confirm you can select "Outreach and Events: Outreach Events" option in "Event listing" field
- [x] assign Tucker to "Office of Public & Intergovernmental Affairs" section and unassign from "Outreach Hub". Create an Event - http://pr1800.ci.cms.va.gov/node/add/event and confirm you can select "Outreach and Events: Outreach Events" option in "Event listing" field
- [x] assign Tucker to "Veterans Affairs" section and unassign from "Office of Public & Intergovernmental Affairs". Create an Event - http://pr1800.ci.cms.va.gov/node/add/event and confirm you can select "Outreach and Events: Outreach Events" option in "Event listing" field

### CODE REVIEW

**Comment by oksana-c** [comment]
also tagging @kevwalsh ^ to validate assumptions in this PR, since access to Outreach Hub will be our topic of conversation this Friday.

File: docroot/modules/custom/va_gov_user/src/Service/UserPermsService.php [hunk 1]
```diff
@@ -237,15 +237,21 @@ public function userAccess($entity_id, $entity_type, AccountInterface $account,
...
       $status_class_name = get_class($scheme->getAccessScheme()->checkEntityAccess($scheme, $entity, $op, $account));
       // Return true if we have our special snowflake Outreach listing node.
-      if ((count($results) > 0) && ($entity->id() === '736')) {
+      // 735 = "Outreach and events" Office node id.
+      // User who are assigned to the following sections are able to access
+      // Outreach Listing node "Outreach and events: Outreach Events" (nid:736):
+      // - Veterans Affairs (tid:5 | section_id:7)
+      // -- Office of Public & Intergovernmental Affairs (tid:6 | section_id:9)
+      // --- Outreach Hub (tid:7 | section_id:4)
+      if (is_array($results) && count($results) > 0 && ($entity->id() === '735')) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_user/src/Service/UserPermsService.php:254]: Does it need to be either 735 or 736?   In the QA tests this had originally  the case for snowflake 736 worked.... makes me question if it needs both.  I still don't understand this snowflake issue as much as i wish I do.

Comment (oksana-c) [reply docroot/modules/custom/va_gov_user/src/Service/UserPermsService.php:254]: let's cover this on Friday

**Comment by oksana-c** [comment]
@howardbrande , thanks for bringing this up. I'm not certain if that is expected behavior. Adding you to Friday call.

**Comment by oksana-c** [comment]
@kevwalsh I updated the owner for the 3 outreach nodes we identified to Outreach Hub on prod per our discussion earlier today.
The latest update to this PR is removing the snowflake treatment which is not needed anymore after data update ^.

**Comment by swirtSJW** [comment]
Nice work on this @oksana-c

**Pull request merged**